asserts/asserts.go

```diff
@@ -872,6 +873,31 @@ func Assemble(headers map[string]interface{}, body, content, signature []byte) (
 	return assemble(headers, body, content, signature)
 }
 
+func checkAuthority(assertType *AssertionType, headers map[string]interface{}) (hasSignatoryID bool, err error) {
+	_, hasSignatoryID = headers["signatory-id"]
+	if _, err := checkNotEmptyString(headers, "authority-id"); err != nil {
+		return false, err
+	}
+	if hasSignatoryID {
+		if _, err := checkNotEmptyString(headers, "signatory-id"); err != nil {
+			return false, err
+		}
+	}
+	return hasSignatoryID, nil
+}
+
+func checkNoAuthority(assertType *AssertionType, headers map[string]interface{}) error {
+	_, ok := headers["authority-id"]
+	if ok {
+		return fmt.Errorf("%q assertion cannot have authority-id set", assertType.Name)
+	}
+	_, ok = headers["signatory-id"]
+	if ok {
```

Comment:
nitpick: it's clear as it is but it would be slightly more concise if the assignments were inlined in the ifs